ddtrace/internal/flare.py

```diff
@@ -0,0 +1,158 @@
+        if uds_path:
+            agent_url = "uds://%s" % uds_path
+        else:
+            agent_url = "http://%s:%s" % (hostname, port)
+    else:
+        # Else if we can't infer anything from the tracer, rely on the defaults.
+        hostname = ddtrace.Tracer.DEFAULT_HOSTNAME
+        port = ddtrace.Tracer.DEFAULT_PORT
+        agent_url = "http://%s:%s" % (hostname, port)
+
+    resp = ping_agent(hostname=hostname, port=port, uds_path=uds_path)
+
+    if isinstance(resp, ddtrace.api.Response):
+        if resp.status == 200:
+            agent_error = None
+        else:
+            agent_error = "HTTP code %s, reason %s, message %s" % (resp.status, resp.reason, resp.msg)
+    else:
+        # There was an exception
+        agent_error = "Exception raised: %s" % str(resp)
+
+    is_venv = in_venv()
+
+    packages_available = {p.project_name: p.version for p in pkg_resources.working_set}
+    integration_configs = {}
+    for module in ddtrace.monkey.PATCH_MODULES:
+        # TODO: this check doesn't work in all cases... we need a mapping
+        #       between the module and the library name.
+        module_available = module in packages_available
+        enabled = ddtrace.monkey.PATCH_MODULES[module]
+        module_instrumented = module in ddtrace.monkey._PATCHED_MODULES
+        module_imported = module in sys.modules
+
+        if enabled:
+            # DEV: integration configs aren't added until the integration module is
+            #      imported.
+            config = ddtrace.config._config.get(module, "N/A")
+        else:
+            config = None
+
+        if module_available:
+            integration_configs[module] = dict(
+                enabled=enabled,
+                instrumented=module_instrumented,
+                module_available=module_available,
+                module_version=packages_available[module],
+                module_imported=module_imported,
+                config=config,
+            )
+        else:
+            # Use N/A here to avoid the additional clutter of an entire
+            # config dictionary for a module that isn't available.
+            integration_configs[module] = "N/A"
+
+    pip_version = packages_available.get("pip", "pip not available")
+
+    return dict(
+        # Timestamp UTC ISO 8601
+        date=str(datetime.datetime.utcnow().isoformat()),
+        # eg. "Linux", "Darwin"
+        os_name=platform.system(),
+        # eg. 12.5.0
+        os_version=platform.release(),
+        is_64_bit=sys.maxsize > 2 ** 32,
+        architecture=platform.architecture()[0],
+        vm=platform.python_implementation(),
+        version=ddtrace.__version__,
+        lang="python",
+        lang_version=platform.python_version(),
+        pip_version=pip_version,
+        in_virtual_env=is_venv,
+        agent_url=agent_url,
+        agent_error=agent_error,
+        env=ddtrace.config.env or "",
+        enabled=formats.asbool(os.getenv("DATADOG_TRACE_ENABLED") or "true"),
```

Comment:
If this is the value to make decision, then I'd suggest yes as we have the real value used by the tracer.

Reply:
Yes we can, and for the `enabled` field it probably makes more sense. The environment variable leads to some other interesting consequences so it might still be worth reporting it separately.